kivy/core/window/__init__.py

```diff
@@ -528,6 +528,21 @@ def _get_system_size(self):
     '''Real size of the window ignoring rotation.
     '''
 
+    def _get_effective_size(self):
+        w, h = self.system_size
+        if platform == 'ios' or self._density != 1:
+            w, h = self.size
+
+        return w,h
+
+    effective_size = AliasProperty(_get_effective_size, None)
+    '''On density=1 and non-ios displays, return system_size, else
+    return scaled / rotated size.
+
```

Comment:
This function would be used internally only and thus it makes sense to keep
it private, i.e have it start using `_`.

On Thu, Jul 16, 2015 at 4:43 PM, Charl P. Botha <notifications@github.com>
wrote:

> In kivy/core/window/__init__.py
> <https://github.com/kivy/kivy/pull/3506#discussion_r34777003>:
>
> > @@ -528,6 +528,21 @@ def _get_system_size(self):
> >      '''Real size of the window ignoring rotation.
> >      '''
> >
> > +    def _get_effective_size(self):
> > +        w, h = self.system_size
> > +        if platform == 'ios' or self._density != 1:
> > +            w, h = self.size
> > +
> > +        return w,h
> > +
> > +    effective_size = AliasProperty(_get_effective_size, None)
> > +    '''On density=1 and non-ios displays, return system_size, else
> > +    return scaled / rotated size.
> > +
>
> My reasons for doing it this way was indeed to keep it DRY (having two
> different places checking the same logic is a bit fragile), and to follow
> the lead of the other properties.
>
> Would you prefer that we rather just remove the property, and then rename
> the function to "get_effective_size()" (without underscore) and use that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kivy/kivy/pull/3506/files#r34777003>.
>